art/utils.py

```diff
@@ -162,13 +162,13 @@ def wrapper(*args, **kwargs):
 # ----------------------------------------------------------------------------------------------------- MATH OPERATIONS
 
 
-def projection(values: np.ndarray, eps: float, norm_p: Union[int, float]) -> np.ndarray:
+def projection(values: np.ndarray, eps: float, norm_p: Union[int, str]) -> np.ndarray:
```

Comment:
Why don't you count `float` here as in other places?

Reply:
I think it just got forgotten, I have added it back for `np.inf`.